### ISSUE TITLE
Unzip Fails for Directory Attributes

### ISSUE THREAD

**Issue by agmsmith** [haikuports/haikuports#2075]
As discussed in https://dev.haiku-os.org/ticket/13866 the currently used unzip in Haiku spews error messages when handling archives with directory attributes.  The easiest fix is to go back to version 6.0, not the unreleased unzip beta 6.10 (looking a the SourceForge file listings https://sourceforge.net/projects/infozip/files/ it wasn't finished).  Yes, it has a higher version number, but doesn't work.

So, how to get the build system to use the older unzip 6.0 recipe?

**Comment by mmlr** [comment]
The 6.10 recipe should be marked broken on all archs, that will remove the corresponding packages. Then the 6.0 recipe should have its revision bumped to ensure that packages for that version are built.

**Comment by fbrosson** [comment]
Should be fixed with
* 2baef7651532f08517b1493e7303db05b39baa7b
* a9ff719e3dd1ce1c51529801023a4a1f75dbdea7

**Comment by agmsmith** [comment]
Thanks!  Will close this once it gets through to the nightly build.

**Comment by agmsmith** [comment]
For future reference, PulkoMandy says "We don't want to ship with 6.0 as-is because it has other problems, in particular with handling japanese characters in zipped filenames (not everyone may notice it, but still)."

**Comment by fbrosson** [comment]
Oops, sorry @pulkomandy. I should have opened a PR instead of merging it directly.
Well, anyway, it seems it was the right thing to do until 6.10 is fixed.

**Comment by pulkomandy** [comment]
> Oops, sorry @pulkomandy. I should have opened a PR instead of merging it directly.
> Well, anyway, it seems it was the right thing to do until 6.10 is fixed.

Replacing a bug with another? Yes, sounds fine...

**Comment by fbrosson** [comment]
Hmm, shall I mark unzip 6.10 as untested (instead of broken)?
This might be a good compromise.
Of course, I can also mark it as tested, if you prefer.

**Comment by pulkomandy** [comment]
On Sun, Jan 07, 2018 at 10:47:09AM +0000, fbrosson wrote:
> Hmm, shall I mark unzip 6.10 as untested (instead of broken)?
> This might be a good compromise.

No, it is not. The code was broken, but so was 6.0 :)

> Of course, I can also mark it as tested, if you prefer.

I updated the recipe to 6.10c23 and backported the fixes from the 6.0
patch (it seems korli worked on 6.0 while I was still preparing the 6.10
port). So we noow have a 6.10 with working file attributes again.

I have also sent the changes (again) to upstream here so maybe they will
be integrated in a future beta:
http://www.info-zip.org/phpBB3/viewtopic.php?f=4&t=377&p=3285#p3285

Additionally I ported the changes from unzip 6.0 to support 64-bit
systems. So the new 6.10c23 recipe should also work fine for 64-bit. I
marked it untested however, because I did not test that.

"Fix bugs, don't hide them"!
